### PR TITLE
Fixed itertuples usage in to_dict

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/24940
Closes https://github.com/pandas-dev/pandas/issues/24939

### CODE REVIEW

**Review by jorisvandenbossche — COMMENTED** [review]
I think the other `itertuples` usages (the same commit that caused this regression also introduced its usage in the 'split' method) are OK? As they don't use the names of the namedtuples?

File: doc/source/whatsnew/v0.24.1.rst [hunk 1]
```diff
@@ -29,8 +29,8 @@ Bug Fixes
 
 **Conversion**
 
--
--
+- Bug in :meth:`DataFrame.itertuples` with ``records`` orient raising an AttributeError when the ``DataFrame`` contained more than 255 columns (:issue:`24939`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.1.rst]: double backticks around AttributeError

**Review by jreback — APPROVED** [review]
minor comments, lgtm.

**Comment by TomAugspurger** [comment]
Switched to using plain tuples for all these.

File: pandas/tests/frame/test_convert_to.py [hunk 2]
```diff
@@ -488,3 +488,17 @@ def test_to_dict_index_dtypes(self, into, expected):
...
+    def test_to_dict_numeric_names(self):
+        # https://github.com/pandas-dev/pandas/issues/24940
+        df = DataFrame({str(i): [i] for i in range(5)})
+        result = set(df.to_dict('records')[0].keys())
+        expected = set(df.columns)
+        assert result == expected
+
+    def test_to_dict_wide(self):
+        # https://github.com/pandas-dev/pandas/issues/24939
+        df = DataFrame({('A_%d' % i): [i] for i in range(256)})
```
Comment (jreback) [inline pandas/tests/frame/test_convert_to.py]: can you use .format here

**Review by jreback — APPROVED** [review]
minor comment

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**